Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/settings/ui/SettingsAdapter.java

```diff
@@ -274,6 +300,54 @@ public void onInputBindingClick(final InputBindingSetting item, final int positi
     dialog.show();
   }
 
+  public void onFilePickerDirectoryClick()
+  {
+    FileBrowserHelper.openDirectoryPicker(mView.getActivity());
+  }
+
+  public void onFilePickerFileClick(SettingsItem item)
+  {
+    mFilePicker = (FilePicker) item;
+    mItem = item;
+
+    FileBrowserHelper.openFilePicker(mView.getActivity(), mFilePicker.getRequestType(), false);
+  }
+
+  public static void onFilePickerConfirmation(String file)
+  {
+    NativeLibrary
+            .SetConfig(SettingsFile.FILE_NAME_DOLPHIN + ".ini", mItem.getSection(), mItem.getKey(),
+                    file);
+    NativeLibrary.ReloadConfig();
+  }
+
+  public static void clearDefaultISO()
+  {
+    NativeLibrary
+            .SetConfig(SettingsFile.FILE_NAME_DOLPHIN + ".ini", Settings.SECTION_INI_CORE,
+                    SettingsFile.KEY_DEFAULT_ISO, "");
+    NativeLibrary.ReloadConfig();
+  }
+
+  public static Set<String> getExtensions()
+  {
+    try
+    {
+      if (mFilePicker.getRequestType() == MainPresenter.REQUEST_SD_FILE)
+      {
+        return SDExtensions;
+      }
+      else
+      {
+        return gameExtensions;
+      }
+    }
+    catch (NullPointerException ex)
```

Comment:
Is there a situation where this gets triggered in practice? Assuming it can happen when calling `getRequestType()`, I feel it would be cleaner to just check if `mFilePicker` is `null`.

Reply:
Yes, this is necessary. FilePickers called from the add directory button and "Open File" menu don't have extensions set in `SettingsFragmentPresenter`. These cases don't mix well with the static variables used in `getExtensions`. There are other cases in Dolphin where we use this method to handle possibly null int too (search for `NullPointerException ex`)